core/src/main/java/com/google/cloud/sql/core/InternalConnectorRegistry.java

```diff
@@ -241,7 +241,7 @@ public static void setApplicationName(String applicationName) {
   }
 
   private Connector getConnector(ConnectionConfig config) {
-    return connectors.computeIfAbsent(config.getCloudSqlInstance(), k -> createConnector(config));
+    return connectors.computeIfAbsent(new ConnectorKey(config), k -> createConnector(config));
```

Comment:
In AlloyDB, we used a string to distinguish. Do we prefer one approach over another here?

https://github.com/GoogleCloudPlatform/alloydb-java-connector/blob/7fc16e7e10df096fc0dd226144e348ae4b5c3866/alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/ConnectionConfig.java#L78-L91

cc @ttosta-google

Reply:
I prefer this approach. Now that we have introduced connection configuration parameters set per-connection configuration like 'targetPrincipal', the InternalConnectorRegistry will need to create a connector for each unique configuration of connector parameters. In this implementation, the InternalConnectorRegistry will automatically create exactly as many connectors as the application needs, usually just 1 connector.

Reply:
I like this approach as well. It is cleaner and easy to add new parameters if needed.

Reply:
@ttosta-google let's add this to AlloyDB in that case. Would you file an issue?

Reply:
Should we do it as part of https://github.com/GoogleCloudPlatform/alloydb-java-connector/issues/273?